Improve: use constants to index into special exit columns in dlgRoomExits

SlySven opened the pull request:
This will make changing/adding the columns easier as part of what was created as the draft (and to be replaced) PR #5308 - that is to be broken up into smaller, easier to digest, chunks! 8-)

~~I also needed to duplicate the column indexes in a second class but I can't see how to the share a single set of index definitions between the `dlgRoomExit` class and the `exitstreewidget` one - the latter is a base class on which the former depends but getting the one to inherit from the other is beyond me...~~*There is a single set of constants which are defined in the `exitstreewidget` but used also by the `dlgRoomExit` one by virtue of making it a friend of the former.*

As it happens I spotted the need for ~~the second set~~ *both classes to have these constants* when I reviewed `(void) ExitsTreeWidget::keyPressEvent(QKeyEvent*)` and spotted that I had not revised it to close the correct persistent editors in the merge commit 83db9f94eda83ede696b9b0ead9b236f3c8ab05b (part of PR #439) seven years ago!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

#### Release post highlight

None - I think - this is an internal tweak to make a subsequent PR easier to follow

##### Edited
Following ff6deef614b3f4791b6951bc4e01e2235e85c3b7 ensure the text squashed into the final commit message for this PR reflects the revised text!

SlySven committed:
Improve: use constants to index into the dlgRoomExits Special Exits columns

This will make changing/adding the columns easier as part of what was
created as the draft (and to be replaced) PR #5308 - that is to be broken
up into smaller, easier to digest, chunks! 8-)

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Bugfix: duplicate column indexes in second class & close correct PerEditors

I can't see how to share a single set of index definitions between the
`dlgRoomExit` class and the `exitstreewidget` one - the latter is a base
class on which the former depends but getting the one to inherit from the
other is beyond me...

As it happens I spotted the need for the second set when I reviewed
`(void) ExitsTreeWidget::keyPressEvent(QKeyEvent*)` and spotted that I had
not revised it to close the correct persistent editors in the merge
commit 83db9f94eda83ede696b9b0ead9b236f3c8ab05b (part of PR #439) seven
years ago!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 commented:
> I also needed to duplicate the column indexes in a second class

Would the "friend" mechanic in C++ help here?

vadi2 reviewed (commented):
Definitely a welcome improvement 👍

SlySven commented:
>> I also needed to duplicate the column indexes in a second class
>>
>Would the "friend" mechanic in C++ help here?

:astonished: Of course - that is what I needed...

SlySven committed:
Revise: make dlgRoomExits a friend of ExitsTreeWidget

This is so that all the (`static const`) `colIndex_Xxxx` values defined in
the latter can also be used in the former (so as to remove the previous
duplication I had to use)!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

demonnic commented:
need to merge development branch back into this PR so it can run the new checks.

SlySven committed:
Merge branch 'development' into Improve_useConstantsToIndexIntoSpecialExitColumnsIn_dlgRoomExits

Needed to pick up the danger.js stuff recently inserted into the CI builds.

Signed-off by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: placate CodeFactor god/daemon...!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>
